modGui/src/org/aion/gui/model/TransactionProcessor.java

```diff
@@ -0,0 +1,270 @@
+        if (dto == null || !dto.validate()) {
+            throw new ValidationException("Invalid transaction request data");
+        }
+        if (dto.estimateValue().compareTo(getBalance(dto.getFrom())) >= 0) {
+            throw new ValidationException("Insufficient funds");
+        }
+        return sendTransactionInternal(dto);
+    }
+
+    private BigInteger getBalance(final String address) {
+        return balanceRetriever.getBalance(address);
+    }
+
+    private TransactionResponseDTO sendTransactionInternal(final SendTransactionDTO dto) {
+        final BigInteger latestTransactionNonce = getLatestTransactionNonce(dto.getFrom());
+        TxArgs txArgs = new TxArgs.TxArgsBuilder()
+                .from(new Address(TypeConverter.toJsonHex(dto.getFrom())))
+                .to(new Address(TypeConverter.toJsonHex(dto.getTo())))
+                .value(dto.getValue())
+                .nonce(latestTransactionNonce)
+                .data(new ByteArrayWrapper(dto.getData()))
+                .nrgPrice(dto.getNrgPrice())
+                .nrgLimit(dto.getNrg())
+                .createTxArgs();
+        final MsgRsp response;
+
+        consoleManager.addLog("Sending transaction", ConsoleManager.LogType.TRANSACTION, ConsoleManager.LogLevel.INFO);
+        response = callApi(api -> api.getTx().sendSignedTransaction(
+                txArgs,
+                new ByteArrayWrapper((accountManager.getAccount(dto.getFrom())).getPrivateKey())
+                )).getObject();
+
+        final TransactionResponseDTO transactionResponseDTO = mapTransactionResponse(response);
+        final int responseStatus = transactionResponseDTO.getStatus();
+        if (!ACCEPTED_TRANSACTION_RESPONSE_STATUSES.contains(responseStatus)) {
+            accountManager.addTimedOutTransaction(dto);
+        }
+        return transactionResponseDTO;
+    }
+
+    private BigInteger getLatestTransactionNonce(final String address) {
+        if(apiIsConnected()) {
+            return callApi(api -> api.getChain().getNonce(Address.wrap(address))).getObject();
+        } else {
+            return BigInteger.ZERO;
+        }
+    }
+
+    private TransactionResponseDTO mapTransactionResponse(final MsgRsp response) {
+        return new TransactionResponseDTO(response.getStatus(), response.getTxHash(), response.getError());
+    }
+
+    private void processTransactionsFromOldestRegisteredSafeBlock() {
+        final Set<String> addresses = accountManager.getAddresses();
+        final Consumer<Iterator<String>> nullSafeBlockFilter = Iterator::remove;
+        final BlockDTO oldestSafeBlock = accountManager.getOldestSafeBlock(addresses, nullSafeBlockFilter);
+        if (oldestSafeBlock != null) {
+            processTransactionsFromBlock(oldestSafeBlock, addresses);
+        }
+    }
+
+    private static int getCores() {
+        int cores = Runtime.getRuntime().availableProcessors();
+        if (cores > 1) {
+            cores = cores / 2;
```

Comment:
this is not true, cause not every cpu support HT, or the HT been diabled in the BIOS.

Reply:
I'm not totally sure why this was coded this way when I inherited the code.  I think here, "getCores" is supposed to mean "get the number of cores this Transaction Processor should use," not "get the number of cores this machine has."

(This value gets passed into a thread-pool executor; it wouldn't make sense to use every core of the machine in a single executor)